chemprop/v2/data/dataloader.py

```diff
@@ -54,7 +82,7 @@ class MolGraphDataLoader(DataLoader):
 
     def __init__(
         self,
-        dataset: _MolGraphDatasetMixin,
+        dataset: MoleculeDataset | ReactionDataset | MulticomponentDataset,
```

Comment:
All three of these `MoleculeDataset | ReactionDataset | MulticomponentDataset` inherit from `_MolGraphDatasetMixin` so the previous type hint should work. I assume you list all three as type hints instead of just the mixin because you want to be more clear for developers?

Reply:
the mixin class is not an actual `torch.Dataset`, it's just, well, a "mixin" that provides some base utilities shared among all three classes. It doesn't actually implement the abstract `torch.Dataset` interface.